libcsm/tests/test_os.py

```diff
@@ -53,6 +55,19 @@ def test_run_command(self) -> None:
             assert isinstance(result.return_code, int)
             assert result.stdout or result.stderr
 
+    def test_run_command_unicode_error(self) -> None:
+        """
+        Assert that UnicodeDecodeErrors are caught by the run_command 
+        and a non-zero return code is recieved.
+        """
+        # test with an invalid utf-8 byte
+        byte_string = b'\x9c'
```

Comment:
Note, this is a CP1252 (windows ostensibly) character oe. 

```
>>> print(b'\x9c'.decode('cp1252'))
œ
```
Should also note that, but to be honest decoding a cp1252 encoded byte as utf8 and getting an exception isn't unexpected.

Reply:
As a caller/user of run_command i'd also expect this not to be a problem and let me sort out the encoding issues on my side not pass up an exception.

Reply:
Made optional.